src/custom/hooks/useGetReceipt.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react'
+import { useActiveWeb3React } from 'hooks/web3'
+import { retry, RetryableError, RetryOptions } from 'utils/retry'
+import { TransactionReceipt } from '@ethersproject/abstract-provider'
+
+const DEFAULT_RETRY_OPTIONS: RetryOptions = { n: 3, minWait: 1000, maxWait: 3000 }
+const RETRY_OPTIONS_BY_CHAIN_ID: { [chainId: number]: RetryOptions } = {
+  // [SupportedChainId.ARBITRUM_ONE]: { n: 10, minWait: 250, maxWait: 1000 },
+  // [SupportedChainId.ARBITRUM_KOVAN]: { n: 10, minWait: 250, maxWait: 1000 },
+}
+
+interface WithCancel {
+  cancel: () => void
+}
+
+interface ReceiptResult extends WithCancel {
+  promise: Promise<TransactionReceipt>
+}
+
+export type GetReceipt = (hash: string) => ReceiptResult
+
+export function useGetReceipt(): GetReceipt {
+  const { chainId, library } = useActiveWeb3React()
+
+  const getReceipt = useCallback<GetReceipt>(
+    (hash) => {
+      if (!library || !chainId) throw new Error('No library or chainId')
```

Comment:
do we really want to throw here? this will break if not caught properly somewhere no? maybe better to return `null`

Reply:
Normally, no-one would use this method until you have a chainId and library. However, what i did, just to not complicate the signature adding the `null` possibility, I moved it to the retry callback, so it will re-attempt automatically